src/boot/efi/cpio.c

```diff
@@ -336,9 +337,24 @@ EFI_STATUS pack_cpio(
                 return EFI_SUCCESS;
         }
 
-        root = LibOpenRoot(loaded_image->DeviceHandle);
-        if (!root)
-                return log_error_status_stall(EFI_LOAD_ERROR, L"Unable to open root directory.");
+        err = uefi_call_wrapper(BS->HandleProtocol, 3, loaded_image->DeviceHandle,
+                                &FileSystemProtocol, (VOID*)&volume);
+        /* It will be unsupported if not vfat. This can happen if the efi was
```

Comment:
why would that be unsupported? Not sure I follow? What does that have to do with vfat vs. ext4?

Reply:
UEFI mandates a version of FAT for the system partition, so usually implementations will only implement support for that.

Reply:
Sure, but the `FileSystemProtocol` EFI protocol is fs independent. So if you EFI environment implements ext4, then the protocol should sitll work.

I think Grub doesn't actually implement ext4 as an UEFI fs though, right? it just does its own thing, loads everything into memory on its own, just exposing the loaded files, but never the backing fs.

So, the comments are confusing here, and the commit msg, because it's not about the choice of backing fs, but the fact that grub simply doesn't implement the file system protocol at *all* for the EFI binaries.

Reply:
I mean, Grub *could* if it wanted implement the FileSystemProtocol stuff, which is after all exactly what this project does: https://github.com/pbatard/efifs

It just chose not to do that.